Extension/src/ghAPI.ts

```diff
@@ -0,0 +1,152 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+'use strict';
+
+import { PackageVersion } from './packageVersion';
+import * as util from './common';
+import * as tmp from 'tmp';
+import * as telemetry from './telemetry';
+import { PlatformInformation } from './platform';
+
+async function parseJsonAtPath(path: string): Promise<any> {
+    try {
+        const exists: boolean = await util.checkFileExists(path);
+        if (exists) {
+            const fileContent: string = await util.readFileText(path);
+            return JSON.parse(fileContent);
+        }
+    } catch (error) {
+    }
+}
+
+interface Asset {
+    name: string;
+    browserDownloadUrl: string;
+}
+
+interface Build {
+    name: string;
+    assets: Asset[];
+}
+
+function isAsset(input: any): input is Asset {
+    return input && input.name && typeof(input.name) === "string" && input.browser_download_url && input.browser_download_url && typeof(input.browser_download_url) === "string"; 
+}
+
+function isBuild(input: any): input is Build {
+    return input && input.name && typeof(input.name) === "string" && isArrayOfAssets(input.assets) && input.assets.length >= 4;
+}
+
+function isArrayOfAssets(input: any): input is Asset[] {
+    return input instanceof Array && input.every(item => isAsset(item));
+}
+
+function isReleaseJson(input: any): input is Build[] {
+    return input && input instanceof Array && input.length !== 0 && input.every(item => isBuild(item));
+}
+
+async function downloadUrlForPlatform(build: Build): Promise<string> {
+    // Get the VSIX name to search for in build
+    const platformInfo: PlatformInformation = await PlatformInformation.GetPlatformInformation();
+    const vsixName: string = function(platformInfo): string {
+        switch (platformInfo.platform) {
+            case 'win32': return 'cpptools-win32.vsix';
+            case 'darwin': return 'cpptools-osx.vsix';
+            default: {
+                switch (platformInfo.architecture) {
+                    case 'x86': return 'cpptools-linux32.vsix';
+                    case 'x86_64': return 'cpptools-linux.vsix';
+                }
+            }
+        }
+    }(platformInfo);
+    if (!vsixName) {
+        return;
+    }
+
+    // Get the URL to download the VSIX, using vsixName as a key
+    const downloadUrl: string = build.assets.find((asset) => {
+        return asset.name === vsixName;
+    }).browserDownloadUrl;
+
+    return downloadUrl;
+}
+
+export async function getTargetBuildURL(updateChannel: string): Promise<string> {
+    const releaseJson: Build[] = await getReleaseJson();
```

Comment:
It's typed as a Build[], so I'd prefer removing "json" from the name of the variable.  Consider calling it `releases` or `builds` (plural because it's an array) instead.